workflows/prognostic_c48_run/emulation/emulation/_monitor/monitor.py

```diff
@@ -0,0 +1,257 @@
+    if key.endswith("_input"):
+        var_key = key[:-6]
+        logger.debug(f"Removed _input with result {var_key} for metadata mapping")
+    elif key.endswith("_output"):
+        var_key = key[:-7]
+        logger.debug(f"Removed _output with result {var_key} for metadata mapping")
+    else:
+        var_key = key
+
+    return var_key
+
+
+def _get_attrs(key: str, metadata: Mapping):
+    key = _remove_io_suffix(key)
+    if key in metadata:
+        meta = dict(**metadata[key])
+        meta = {k: json.dumps(v) for k, v in meta.items()}
+    else:
+        logger.debug(f"No metadata found for {key}... skipping")
+        meta = {}
+
+    return meta
+
+
+def _convert_to_quantities(state, metadata):
+
+    quantities = {}
+    for key, data in state.items():
+        data = np.squeeze(data.astype(np.float32))
+        data_t = data.T
+        dims = DIMS_MAP[data.ndim]
+        attrs = _get_attrs(key, metadata)
+        units = attrs.pop("units", "unknown")
+        quantities[key] = Quantity(data_t, dims, units)
+        # Access to private member could break TODO: Quantity kwarg for attrs?
+        quantities[key]._attrs.update(attrs)
+
+    return quantities
+
+
+def _convert_to_xr_dataset(state, metadata):
+
+    dataset = {}
+    for key, data in state.items():
+        data = np.squeeze(data.astype(np.float32))
+        data_t = data.T
+        dims = DIMS_MAP[data.ndim]
+        attrs = _get_attrs(key, metadata)
+        attrs["units"] = attrs.pop("units", "unknown")
+        dataset[key] = xr.DataArray(data_t, dims=dims, attrs=attrs)
+
+    return xr.Dataset(dataset)
+
+
+def _translate_time(time):
+
+    # list order is set by fortran from variable Model%jdat
+    year = time[0]
+    month = time[1]
+    day = time[2]
+    hour = time[4]
+    min = time[5]
+    datetime = cftime.DatetimeJulian(year, month, day, hour, min)
+    logger.debug(f"Translated input time: {datetime}")
+
+    return datetime
+
+
+def _create_nc_path():
+
+    nc_dump_path = os.path.join(os.getcwd(), "netcdf_output")
+    if not os.path.exists(nc_dump_path):
+        os.makedirs(nc_dump_path, exist_ok=True)
+
+    return nc_dump_path
+
+
+def _store_netcdf(state, time, nc_dump_path, metadata):
+
+    logger.debug(f"Model fields: {list(state.keys())}")
+    logger.info(f"Storing state to netcdf on rank {MPI.COMM_WORLD.Get_rank()}")
+    ds = _convert_to_xr_dataset(state, metadata)
+    rank = MPI.COMM_WORLD.Get_rank()
+    coords = {"time": time, "tile": rank}
+    ds = ds.assign_coords(coords)
+    filename = f"state_{time.strftime(TIME_FMT)}_{rank}.nc"
+    out_path = os.path.join(nc_dump_path, filename)
+    ds.to_netcdf(out_path)
+
+
+def _store_zarr(state, time, monitor, metadata):
+
+    logger.info(f"Storing zarr model state on rank {MPI.COMM_WORLD.Get_rank()}")
+    logger.debug(f"Model fields: {list(state.keys())}")
+    state = _convert_to_quantities(state, metadata)
+    state["time"] = time
+    monitor.store(state)
+
+
+class Config:
```

Comment:
I would rename this to `StorageHook` or something similar since it has methods like `microphysics` and `store` that seem unrelated to configuration.